### DISCUSSION TITLE
Improve eager_loading performance

### DISCUSSION THREAD

**Discussion by exseniorastronaut** [aarondl/sqlboiler#889] [Ideas]
It would be nice if somehow we could eager_load non recursive relations on their own go-routine to speed up the process. For example if you are eager_loading 6,7 relations this would improve the performance significally.

**Comment by aarondl** [comment]
Perhaps this can be part of the eager loading rewrite which is sorely needed.

**Reply by ccakes** [reply]
Is it possible at all to preload using SQL `JOIN`s instead of the additional queries? This is how ORMs in other languages tend to handle preloading,  I'm new to Go but any that do support a form of preloading do so using multiple queries.

**Reply by aarondl** [reply]
It most likely is. But note that it's not the case for to-many relationships. Most ORMs (things like ActiveRecord included) will do separate queries instead of left joins because otherwise you get the difficult-to-deal with sql output:

```
car happy
red 2
red 5
red 6
```
